modules/crud/sql/describe.go

```diff
@@ -2,27 +2,56 @@ package sql
 
 import (
 	"context"
-	"fmt"
 
 	"github.com/spaceuptech/space-cloud/utils"
 )
 
 // DescribeTable return a description of sql table & foreign keys in table
 // NOTE: not to be exposed externally
-func (s *SQL) DescribeTable(ctx context.Context, project, col string) ([]utils.FieldType, []utils.ForeignKeysType, error) {
-	fields, err := s.getDescribeDetails(ctx, col)
+func (s *SQL) DescribeTable(ctx context.Context, project, col, dbType string) ([]utils.FieldType, []utils.ForeignKeysType, error) {
+	fields, err := s.getDescribeDetails(ctx, col, dbType)
 	if err != nil {
 		return nil, nil, err
 	}
-	foreignKeys, err := s.getForeignKeyDetails(ctx, project, col)
+	foreignKeys, err := s.getForeignKeyDetails(ctx, project, col, dbType)
 	if err != nil {
 		return nil, nil, err
 	}
 	return fields, foreignKeys, nil
 }
 
-func (s *SQL) getDescribeDetails(ctx context.Context, col string) ([]utils.FieldType, error) {
-	rows, err := s.client.Queryx("DESCRIBE " + col)
+func (s *SQL) getDescribeDetails(ctx context.Context, col, dbType string) ([]utils.FieldType, error) {
+	queryString := ""
+	if utils.DBType(dbType) == utils.MySQL {
+		queryString = "DESCRIBE " + col
+	} else {
+		queryString = `SELECT  
+		f.attnum AS "Default",  
+		f.attnum AS "Extra",
+		f.attname AS "Field",  
+		pg_catalog.format_type(f.atttypid,f.atttypmod) AS "Type",  
+		CASE  
+			WHEN p.contype = 'p' THEN 'PRI'  
+			WHEN p.contype = 'u' THEN 'UNI'
+			ELSE 'f'  
+		END AS "Key",
+		CASE
+			WHEN f.attnotnull = 't' THEN 'YES'
+			ELSE 'NO'
+		END AS "Null"
+	FROM pg_attribute f  
+		JOIN pg_class c ON c.oid = f.attrelid  
+		JOIN pg_type t ON t.oid = f.atttypid  
+		LEFT JOIN pg_attrdef d ON d.adrelid = c.oid AND d.adnum = f.attnum  
+		LEFT JOIN pg_namespace n ON n.oid = c.relnamespace  
+		LEFT JOIN pg_constraint p ON p.conrelid = c.oid AND f.attnum = ANY (p.conkey)  
+		LEFT JOIN pg_class AS g ON p.confrelid = g.oid  
+	WHERE c.relkind = 'r'::char    
+		AND c.relname = '` + col + `'  -- Replace with table name  
```

Comment:
take col as `:col`. And remove the table name

Reply:
I didn't get u, what's u want to say

Reply:
it is not possible here, i have tried
because of sql describe

Reply:
Do this. Make an array of string. In case of mysql it will be empty, in case of Postgres it would have the database and col. Then spread this array while querying.

Reply:
MySQL statements will have `?` while Postgres will have `$1`